crates/biome_js_analyze/src/lint/nursery/no_unnecessary_template_expression.rs

```diff
@@ -0,0 +1,301 @@
+use crate::JsRuleAction;
+use biome_analyze::{
+    Ast, FixKind, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsLiteralExpression, AnyJsTemplateElement, JsTemplateExpression,
+};
+use biome_rowan::{AstNode, AstNodeList, BatchMutationExt};
+
+declare_lint_rule! {
+    /// Disallow unnecessary template expressions.
+    ///
+    /// This rule reports template literals that contain only a single interpolation or string literal,
+    /// and where the template literal would be equivalent to a simpler expression.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// const ab = `${'a'}${'b'}`;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const wrappedText = `${text}`;
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// const stringWithNumber = `${'1 + 1 = '}${2}`;
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// const ab = 'ab';
+    /// ```
+    ///
+    /// ```js
+    /// const wrappedText = text;
+    /// ```
+    ///
+    /// ```js
+    /// const stringWithNumber = '1 + 1 = 2';
+    /// ```
+    ///
+    /// ```js
+    /// const text = `Hello, ${name}!`;
+    /// ```
+    ///
+    /// ```js
+    /// const multiline = `line 1
+    /// line 2`;
+    /// ```
+    ///
+    /// ```js
+    /// const withQuotes = `It's "nice"`;
+    /// ```
+    ///
+    pub NoUnnecessaryTemplateExpression {
+        version: "2.1.0",
+        name: "noUnnecessaryTemplateExpression",
+        language: "ts",
+        sources: &[RuleSource::EslintTypeScript("no-unnecessary-template-expression").same()],
```

Comment:
This should be `inspired()` because the original rule does more. The original rule:
- uses type information, and this rule doesn't 
- works for temple literal types